include/at_params.h

```diff
@@ -0,0 +1,253 @@
+ */
+int at_params_list_init(struct at_param_list *list,
+			     size_t max_params_count);
+
+
+/**
+ * @brief Clear/reset all parameter types and values.
+ *
+ * All parameter types and values are reset to default values.
+ *
+ * @param[in] list Parameter list to clear.
+ */
+void at_params_list_clear(struct at_param_list *list);
+
+
+/**
+ * @brief Free a list of parameters.
+ *
+ * First the list is cleared. Then the list and its elements are deleted.
+ *
+ * @param[in] list Parameter list to free.
+ */
+void at_params_list_free(struct at_param_list *list);
+
+
+/**
+ * @brief Clear/reset a parameter type and value.
+ *
+ * @param[in] list   Parameter list.
+ * @param[in] index  Parameter index to clear.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int at_params_clear(struct at_param_list *list, size_t index);
+
+
+/**
+ * @brief Add a parameter in the list at the specified index and assign it a
+ * short value.
+ *
+ * If a parameter exists at this index, it is replaced.
+ *
+ * @param[in] list      Parameter list.
+ * @param[in] index     Index in the list where to put the parameter.
+ * @param[in] value     Parameter value.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int at_params_short_put(const struct at_param_list *list, size_t index,
+			u16_t value);
+
+
+/**
+ * @brief Add a parameter in the list at the specified index and assign it an
+ * integer value.
+ *
+ * If a parameter exists at this index, it is replaced.
+ *
+ * @param[in] list      Parameter list.
+ * @param[in] index     Index in the list where to put the parameter.
+ * @param[in] value     Parameter value.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int at_params_int_put(const struct at_param_list *list, size_t index,
+			u32_t value);
+
+
+/**
+ * @brief Add a parameter in the list at the specified index and assign it a
+ * string value.
+ *
+ * The parameter string value is copied and added to the list as a
+ * null-terminated string. If a parameter exists at this index, it is replaced.
+ *
+ * @param[in] list    Parameter list.
+ * @param[in] index   Index in the list where to put the parameter.
+ * @param[in] str     Pointer to the string value.
+ * @param[in] str_len Number of characters of the string value @p str.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int at_params_string_put(const struct at_param_list *list,
+			size_t index, const char *str,
+			size_t str_len);
+
+
+/**
+ * @brief Get the size of a given parameter (in bytes).
+ *
+ * A missing parameter has a size of '0'.
+ *
+ * @param[in] list    Parameter list.
+ * @param[in] index   Parameter index in the list.
+ * @param[out] len    Length of the parameter in bytes or the string length.
```

Comment:
What does the "or" mean here? 
Can it return either or, or are we saying the same thing in two different ways?

Reply:
We are saying the same thing in two different ways here. It would be sufficient to remove "or the string length" at the end of the sentence.

Reply:
fixed